### PR TITLE
add support for X.509

### PR DESCRIPTION
**Pull request by burtbeckwith**
Fixes https://github.com/micronaut-projects/micronaut-security/issues/165

Still needs docs.

### CODE REVIEW

**Review by sdelamo — CHANGES_REQUESTED** [review]
Looks good. It misses documentation.

File: security/src/main/java/io/micronaut/security/x509/X509AuthenticationFetcher.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+     * if one is present in the request.
+     *
+     * @param request the request
+     * @return the authentication if the certificate exists and contains a valid name
+     */
+    protected Optional<Authentication> createAuthentication(HttpRequest<?> request) {
+        Optional<Certificate> optionalCertificate = request.getCertificate();
+        if (optionalCertificate.isPresent()) {
+            Certificate certificate = optionalCertificate.get();
+            if (certificate instanceof X509Certificate) {
```
Comment (aaronzirbes) [inline security/src/main/java/io/micronaut/security/x509/X509AuthenticationFetcher.java:83]: This should support subclasses of `X509Certificate`
```java
X509Certificate.class.isAssignableFrom(certificate.getClass())
```

**Review by aaronzirbes — COMMENTED** [review]
This feature would be great to have!

File: security/src/main/java/io/micronaut/security/x509/X509AuthenticationFetcher.java (hunk 1 above)
Comment (jameskleeh) [reply security/src/main/java/io/micronaut/security/x509/X509AuthenticationFetcher.java:83]: A subclass of X509 is an instance of X509

**Comment by graemerocher** [comment]
tests and check style are failing

**Review by sdelamo — APPROVED** [review]
@burtbeckwith  I have added small changes. Let me know if you have any concerns with these changes. If you don't I will merge the PR.

**Pull request merged**